WordPress/gutenberg
# [Block Library - Post Terms]: Fix handling for low privileged users

## Description
Part of: #26573

Currently when a user with no right permissions to edit `terms` has a broken experience with `Post Terms` blocks (`Post Tags, Post Categories`) due to the permissions check happening at the REST API.

This PR fixes that by changing the fetching of the terms with `context:view`. This block doesn't update any term so it purely `readonly`.



## Testing instructions
- With an admin user insert `Post Tags, Post Categories` and observe that everything works as before.
- With a user whose role hasn't the right permissions(like author, contributor etc...), insert these blocks and observe that are shown properly.
- Also test the case when the value of these blocks is empty and that it updates when we make changes to the post tags or categories by the `Post Inspector controls` (sidebar).

## Notes
- I have also refactored the previous used hook to fetch the data in one call and be responsible for plain data and not React elements (`useTermLinks` -> `usePostTerms`).

## Checklist:
